packages/curvenote/src/check/runner.ts

```diff
@@ -0,0 +1,123 @@
+import path from 'node:path';
+import chalk from 'chalk';
+import {
+  KNOWN_IMAGE_EXTENSIONS,
+  createTempFolder,
+  findCurrentProjectAndLoad,
+  getFileContent,
+  loadProjectFromDisk,
+} from 'myst-cli';
+import { incrementOptions } from 'simple-validators';
+import type { ISession } from '../session/types.js';
+import { CheckStatus } from './types.js';
+import type { Check, CheckDefinition, CheckInterface, CheckReport, CheckResult } from './types.js';
+import { validateCheck } from './validators.js';
+
+export async function runChecks(
+  session: ISession,
+  file: string,
+  checks: Check[],
+  checkDefinitions: CheckInterface[],
+): Promise<CheckReport> {
+  const opts = { property: 'checks', messages: {} };
+  if (!['.ipynb', '.md', '.tex'].includes(path.extname(file))) {
```

Comment:
Here we probably also want to support jats/meca/docx(?) - but I think those should likely be all loaded to mdast here, then the `check` functions don't really care about the source format, only the mdast.

That might be too idealistic though... we will have to see as we implement more real checks against real files.